server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -279,17 +279,32 @@ export class AssetService {
         /**
          * Serve file viewer on the web
          */
-        if (query.isWeb) {
+        if (query.isWeb && asset.mimeType == 'image/gif') {
+          res.set({
+            'Content-Type': asset.mimeType,
+          });
+
+          if (await processETag(asset.originalPath, res, headers)) {
+            return;
+          }
+          await fs.access(asset.originalPath, constants.R_OK | constants.W_OK);
+          fileReadStream = createReadStream(asset.originalPath);
+
+          return new StreamableFile(fileReadStream);
+        } else if (query.isWeb) {
```

Comment:
Man, all of these situations are basically the same:
- Some conditions
- processETag
- fs.access check
- create read stream
- return streamable file

We should probably refactor all the if/else statements to just set a `filepath` variable, and then do the above steps _once_ with the path (or throw an error if the path is empty).  You don't need to do that, I'm just ranting btw.

It looks like your "return the original file" code is exactly the same as another condition further down. Can you update that condition instead of adding a new one? (`if((!query.isThumb && allowOriginalFile) || (query.isWeb && asset.mimeType === 'image/gif'))`

Reply:
Somehow I had missed that block further down... Added it there now. :)